k-distribution/include/builtin/domains.k

```diff
@@ -980,6 +987,10 @@ module K-IO
                  | "#systemResult" "(" Int /* exit code */ "," String /* stdout */ "," String /* stderr */ ")"
 
   syntax K ::= #logToFile(String, String) [function, functional, hook(IO.log), impure, returnsUnit]
+  // #tempFilename ( prefix:String, suffix:String):IOString - creates a temporary file in the default system location
+  syntax IOString ::= #tempFilename(String, String)  [function, functional, hook(IO.tempFilename), impure]
```

Comment:
this is not secure and is subject to the well known time-of-check-time-of-use fault. Please make the following function instead:
```
syntax IOFile ::= IOError | tempFile(path: String, fd: Int)
syntax IOFile ::= #mkstemp(String) [function, hook(IO.mkstemp)]
```

This function should create the temporary filename and open it atomically, then return the file descriptor and the path that was opened. It should return an IOError upon failure.

Reply:
I've looked over Java and Ocaml libraries, and they have both options available. I added both options here too.
I did some minimal testing for `mkstemp`, not sure if this is the correct behavior you were thinking of.

Reply:
I'm aware that for various historical reasons both options are available in most programming languages. That is not a justification to make both options available in K, however.